explorer.go

```diff
@@ -139,19 +176,62 @@ func (s APIEndpoint) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 
 	resp, err := c.Get(skycoinURL)
+	var cachedResponse []byte
+	var cachedResponseFound bool
 	if err != nil {
-		msg := "Request to skycoin node failed"
-		log.Println("ERROR:", msg, skycoinURL)
-		http.Error(w, msg, http.StatusInternalServerError)
-		return
+		// Cancel the operation, but not if a previous cached value is available
+		cachedData, found := cacheMgr.Get(s.ExplorerPath)
+		cachedResponseFound = found
+
+		if cachedResponseFound {
+			cachedResponse = cachedData.([]byte)
+		} else {
+			msg := "Request to skycoin node failed"
+			log.Println("ERROR:", msg, skycoinURL)
+			http.Error(w, msg, http.StatusInternalServerError)
+			return
+		}
 	}
 
-	defer resp.Body.Close()
+	var responseBody io.Reader
+
+	if err != nil {
+		// Simulate a correct answer and use the previously saved one
+		w.WriteHeader(200)
+		responseBody = bytes.NewReader(cachedResponse)
```

Comment:
This only runs if the connection failed, but was possible to find data in the cache.